tests/std/tests/P0009R18_mdspan_extents/test.cpp

```diff
@@ -0,0 +1,294 @@
+        return Int{1};
+    }
+};
+
+struct NonConvertibleToAnything {};
+
+template <class T>
+constexpr void check_implicit_conversion(T); // not defined
+
+template <class T, class... Args>
+concept NotImplicitlyConstructibleFrom =
+    constructible_from<T, Args...> && !requires(Args... args) { check_implicit_conversion<T>({args...}); };
+
+template <class IndexType, size_t... Extents, size_t... Indices>
+constexpr void do_check_members(index_sequence<Indices...>) {
+    using Ext = extents<IndexType, Extents...>;
+
+    // Each specialization of extents models regular and is trivially copyable
+    static_assert(regular<Ext>);
+    static_assert(is_trivially_copyable_v<Ext>);
+
+    // Check member types
+    static_assert(same_as<typename Ext::index_type, IndexType>);
+    static_assert(same_as<typename Ext::size_type, make_unsigned_t<IndexType>>);
+    static_assert(same_as<typename Ext::rank_type, size_t>);
+
+    // Check static observers
+    static_assert(Ext::rank() == sizeof...(Extents));
+    static_assert(Ext::rank_dynamic() == ((Extents == dynamic_extent) + ...));
+    static_assert(((Ext::static_extent(Indices) == Extents) && ...));
+
+    // Check noexceptness of static observers
+    static_assert(noexcept(Ext::rank()));
+    static_assert(noexcept(Ext::rank_dynamic()));
+    static_assert(noexcept(Ext::static_extent(0)));
+
+    // Check default constructor
+    Ext ext;
+    static_assert(is_nothrow_default_constructible_v<Ext>);
+
+    // Check 'extent' observer
+    assert((((ext.extent(Indices) == Extents && Extents != dynamic_extent) || ext.extent(Indices) == 0) && ...));
+
+    using OtherIndexType = conditional_t<is_signed_v<IndexType>, long long, unsigned long long>;
+
+    { // Check construction from other extents
+        using Ext2 = extents<OtherIndexType, Extents...>;
+        Ext2 ext2{ext};
+        assert(((ext.extent(Indices) == ext2.extent(Indices)) && ...));
+        assert(ext == ext2);
+        static_assert(is_nothrow_constructible_v<Ext2, Ext>);
+        // Other tests are defined in 'check_construction_from_other_extents' function
+    }
+
+    { // Check construction from extents pack
+        using Ext2 = extents<OtherIndexType, Extents...>;
+        Ext2 ext2{ext.extent(Indices)...};
+        assert(((ext.extent(Indices) == ext2.extent(Indices)) && ...));
+        assert(ext == ext2);
+        static_assert(is_nothrow_constructible_v<Ext2, Ext>);
+        // Other tests are defined in 'check_construction_from_extents_pack' function
+    }
+
+    { // Check construction from array and span
+        using Ext2 = extents<OtherIndexType, Extents...>;
+
+        auto arr = to_array<IndexType>({ext.extent(Indices)...});
+        Ext ext2a{arr};
+        assert(((ext.extent(Indices) == ext2a.extent(Indices)) && ...));
+        assert(ext == ext2a);
+        static_assert(is_nothrow_constructible_v<Ext2, decltype(arr)>);
+
+        span s{arr};
+        Ext ext2b{s};

```

Comment:
:warning: These are constructing `Ext`, but it appears that `Ext2` was intended, based on their names, and the `is_nothrow_constructible_v<Ext2, MEOW>` lines.